sympy/physics/mechanics/_pathway.py

```diff
@@ -0,0 +1,253 @@
+"""Implementations of pathways for use by actuators.
+
+Notes
+=====
+
+This module is experimental and so is named with a leading underscore to
+indicate that the API is not yet stabilized and could be subject to breaking
+changes.
+
+"""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+from sympy.core.backend import S, Symbol
+from sympy.core.expr import Expr
+from sympy.physics.mechanics import Point
+from sympy.physics.vector import Vector, dynamicsymbols
+
+
+__all__ = ['LinearPathway']
+
+
+class PathwayBase(ABC):
+    """Abstract base class for all pathway classes to inherit from.
+
+    Notes
+    =====
+
+    Instances of this class cannot be directly instantiated by users. However,
+    it can be used to created custom pathway types through subclassing.
+
+    """
+
+    def __init__(
+        self,
+        attachments: tuple[Point, ...],
+    ) -> None:
+        """Initializer for ``PathwayBase``."""
+        self.attachments = attachments
+
+    @property
+    def attachments(self) -> tuple[Point, ...]:
+        """The pair of points defining a pathway's ends."""
+        return self._attachments
+
+    @attachments.setter
+    def attachments(self, attachments: tuple[Point, ...]) -> None:
+        if not iter(attachments):
```

Comment:
You know that this does raise an error, if the object is not iterable? You can just use the `len` check directly or use `iterable` from `sympy.utilities`

Reply:
Good spot. This is not doing what I intended it to do. I will change.

Reply:
Addressed in https://github.com/sympy/sympy/pull/24914/commits/35fc2850aaeeb13ccd04021e649196f5f86783dd.

Reply:
Just a little warning, though I would not expect `Matrix` being passed here:
```
from sympy import Matrix
from sympy.utilities.iterables import iterable
from collections.abc import Iterable

print(isinstance(Matrix(), Iterable))  # prints False
print(iterable(Matrix()))  # prints True
```

Reply:
It looks like it's not possible to make a `Matrix` of `Point` instances. A `Point` will be sympified into a `Symbol` inside of `Matrix`:

```python
>>> P = Point('P')
>>> type(P)
<class 'sympy.physics.vector.point.Point'>
>>> M = Matrix([P])
>>> M[0]
P
>>> type(M[0])
<class 'sympy.core.symbol.Symbol'>
```